core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -592,8 +592,14 @@ private[spark] class BlockManager(
     val locations = Random.shuffle(master.getLocations(blockId))
     for (loc <- locations) {
       logDebug(s"Getting remote block $blockId from $loc")
-      val data = blockTransferService.fetchBlockSync(
-        loc.host, loc.port, loc.executorId, blockId.toString).nioByteBuffer()
+      val data = try {
+        blockTransferService.fetchBlockSync(
+          loc.host, loc.port, loc.executorId, blockId.toString).nioByteBuffer()
+      } catch {
+        case e: Throwable =>
+          logWarning(s"Exception during getting remote block $blockId from $loc", e)
```

Comment:
Thanks @srowen and @CodingCat for comments!
* If i am understanding correctly,the `doGetRemote` here will return `None` when fetched all the same block is null, and all the method which called the `doGetRemote` will handle the `None` case  and throw exception when necessary,so i think it's safe here to catch the exception
* `fetchBlockSync` just call `fetchBlocks` to fetch the block,so i think it's the same we catch exception here.